### DISCUSSION TITLE
is it possible to handle errors when adding record with existing key?

### DISCUSSION THREAD

**Discussion by jacekmamot** [assuncaocharles/ngx-indexed-db#347] [Q&A] [answered]
when I add object with existing key I should be able to handle error (object can not be added)
I want to handle error like in example below
this.dbService

> .add(objectStore, object).subscribe(
{
next: (result) => { console.debug(result); },
error: (error) => { console.error(error); }
});`

It seems that subscribe method supports only positive scenario (next)
I checked code of NgxIndexedDBService.add method and it looks like only request.onsuccess is defined
Am I right?
Can this be changed?

**Comment by aparzi** [comment, marked as answer]
Hi @jacekmamot,
I'll take a look asap and possibly make a change at this point.
I keep you updated

**Comment by aparzi** [comment]
Hi @jacekmamot,
your request is processing with PR #349
